### ISSUE TITLE
IDE fails to compile and issues no error if vec2 is specified as ? instead of (?,?)

### ISSUE THREAD

**Issue by keenancrane** [penrose/penrose#1295]
**Describe the bug**

An common mistake is writing
```
vec2 x = ?
```
rather than
```
vec2 = (?,?)
```
Currently, the IDE fails to report any error if the former is written.  Instead, compile just silently fails, and nothing shows up on the screen.  This is quite tricky to debug if you don't realize the problem is the `vec2 x = ?` line.

**To Reproduce**

Here's a complete Style example—no Domain or Substance program is needed (just leave them empty):

```
canvas {
    width = 200
    height = 200
}

global {

    -- if you write this, compile fails but the
    -- IDE doesn't issue any kind of error
    vec2 p = ?
    
    -- this is how it should be
    -- vec2 p = (?,?)

    shape debug = Circle {
        center: p
        r: 20
        fillColor: rgba(1,0,0,1)
    }
}
```

**Expected behavior**
The IDE should report an error.

**Screenshots**
This is how things look right after hitting "compile."  Nothing drawn, and no error reported; the program remains uncompiled in the IDE.

<img width="1675" alt="image" src="https://user-images.githubusercontent.com/3604525/216436626-fbb4ee8f-7e93-4558-88b3-ee83a758dbf1.png">

**Desktop (please complete the following information if applicable):**

Mac OS X / Chrome

**Additional context**

Thanks!

**Comment by liangyiliang** [comment]
I think this bug was two-fold.

First, the `vec2` type declaration in `vec2 p = ?` is not checked at all. Then, `p` is just one floating-point value `?`. We have not addressed this problem.

Second, `debug = Circle { center: p }` did not throw an error due to shape properties not being type-checked. This problem has been addressed in #1282 which introduces a simple type checker for shape properties. Similarly, after the function-as-data PR (#1352) function arguments have also been type-checked.

To summarize, in all (?) places where `p` is used, `p`'s type is checked, so there should not be silent errors anymore.

@wodeni
